scripts/datacube.py

```diff
@@ -397,24 +397,18 @@ def merge_datasets(ds_sen2, ds_sen1, da_dem):
     - da_dem (xr.DataArray): xarray DataArray for Copernicus DEM data.
 
     Returns:
-    - xr.Dataset: Merged xarray Dataset.
+    - xr.Dataset: Merged xarray Dataset containing valid data region.
     """
-    # print(
-    #     "Platform variables (S2, S1, DEM): ",
-    #     ds_sen2.platform.values,
-    #     ds_sen1.platform.values,
-    #     da_dem.platform.values,
-    # )
-    # ds_sen2 = ds_sen2.drop(["platform", "constellation"])
-    # ds_sen1 = ds_sen1.drop(["platform", "constellation"])
-    # da_dem = da_dem.drop(["platform"])
-
+    # Merging the datasets as before
     ds_merge = xr.merge([ds_sen2, ds_sen1, da_dem], compat="override")
-    print("Merged dataset: ", ds_merge)
-    print(
-        "Time variables (S2, merged): ", ds_sen2.time.values, ds_merge.time.values
-    )  # ds_sen1.time.values, da_dem.time.values
-    return ds_merge
+
+    # Define a mask where all variables have valid data (not NaN)
+    valid_mask = ds_merge.notnull().all(dim=["x", "y"])
```

Comment:
This is not necessary, the problem was with rounding errors on the WGS84 bbox. See https://github.com/Clay-foundation/model/pull/43